python/lsst/resources/s3utils.py

```diff
@@ -11,7 +11,10 @@
 
 from __future__ import annotations
 
+from unittest import TestCase
```

Comment:
The normal way of dealing with the need to import some code that is only used for type checking is to do something like:

```python
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from unittest import TestCase
```